File: Java/Fibonacci.java
public class Fibonacci {
    public static long fibI(int n) {
        long a = 0, b = 1, t;
        for (int i = 0; i < n; i++) { t = a + b; a = b; b = t; }
        return a;
    }
    public static long fibR(int n) {
        if (n <= 1) return n;
        return fibR(n - 1) + fibR(n - 2);
    }
    public static void main(String[] args) {
        int n = 30;
        long s = System.nanoTime(); fibI(n); long e = System.nanoTime();
        System.out.println("Java Iterativo: " + (e - s) / 1e6 + " ms");
        s = System.nanoTime(); fibR(n); e = System.nanoTime();
        System.out.println("Java Recursivo: " + (e - s) / 1e6 + " ms");
    }
}

File: python/fibonacci.py
import time

def fib_i(n):
    a, b = 0, 1
    for _ in range(n): a, b = b, a + b
    return a

def fib_r(n):
    if n <= 1: return n
    return fib_r(n - 1) + fib_r(n - 2)

n = 30
s = time.perf_counter(); fib_i(n); e = time.perf_counter()
print(f"Python Iterativo: {(e - s) * 1000:.4f} ms")
s = time.perf_counter(); fib_r(n); e = time.perf_counter()
print(f"Python Recursivo: {(e - s) * 1000:.4f} ms")